evm/deploy-contract.js

```diff
@@ -100,43 +161,95 @@ async function deploy(options, chain) {
+        case 'create2': {
+            salt = salt || contractName;
+            printInfo(`${contractName} deployment salt`, salt);
 
-    if (!create3Deployer) {
-        throw new Error(`Create3 deployer does not exist on ${chain.name}.`);
-    }
+            constAddressDeployer = contracts.ConstAddressDeployer?.address;
+
+            if (!constAddressDeployer) {
+                throw new Error(`ConstAddressDeployer deployer does not exist on ${chain.name}.`);
+            }
+
+            const contractAddress = await predictContractConstant(constAddressDeployer, wallet, implementationJson, salt);
+            printInfo(`${contractName} deployer will be deployed to`, contractAddress);
+            break;
+        }
 
-    const contractAddress = await getCreate3Address(create3Deployer, wallet.connect(provider), salt);
-    printInfo(`${contractName} will be deployed to`, contractAddress);
+        case 'create3': {
+            salt = options.salt || contractName;
+            printInfo(`${contractName} deployment salt`, salt);
+
+            create3Deployer = contracts.Create3Deployer?.address;
+
+            if (!create3Deployer) {
+                throw new Error(`Create3 deployer does not exist on ${chain.name}.`);
+            }
+
+            const contractAddress = await getCreate3Address(create3Deployer, wallet.connect(provider), salt);
+            printInfo(`${contractName} will be deployed to`, contractAddress);
+            break;
+        }
+    }
 
     if (!yes) {
         console.log('Does this match any existing deployments?');
         const anwser = readlineSync.question(`Proceed with deployment on ${chain.name}? ${chalk.green('(y/n)')} `);
         if (anwser !== 'y') return;
     }
 
-    const contract = await deployCreate3(
-        create3Deployer,
-        wallet.connect(provider),
-        implementationJson,
-        constructorArgs,
-        salt,
-        gasOptions,
-        verifyOptions,
-    );
+    let contract;
+
+    switch (deployMethod) {
+        case 'create': {
+            contract = await deployContract(wallet, implementationJson, [], gasOptions, verifyOptions);
+            break;
+        }
+
+        case 'create2': {
+            contract = await deployCreate2(constAddressDeployer, wallet, implementationJson, [], salt, gasOptions.gasLimit, verifyOptions);
+
+            contractConfig.salt = salt;
+            printInfo(`${chain.name} | ConstAddressDeployer:`, constAddressDeployer);
+            break;
+        }
+
+        case 'create3': {
+            contract = await deployCreate3(
+                create3Deployer,
+                wallet.connect(provider),
+                implementationJson,
+                constructorArgs,
+                salt,
+                gasOptions,
+                verifyOptions,
+            );
+
+            contractConfig.salt = salt;
+            printInfo(`${chain.name} | Create3Deployer:`, create3Deployer);
```

Comment:
```suggestion
            printInfo(`${chain.name} | Create3Deployer`, create3Deployer);
```
`:` is added automatically